connectivity/check/deployment.go

```diff
@@ -763,8 +977,40 @@ func (ct *ConnectivityTest) validateDeployment(ctx context.Context) error {
 	return nil
 }
 
+// Validate that srcPod can query the DNS server on dstPod successfully
+func (ct *ConnectivityTest) waitForPodDNS(ctx context.Context, srcPod, dstPod Pod) error {
+	ct.Logf("⌛ [%s] Waiting for pod %s to reach DNS server on %s pod...", ct.client.ClusterName(), srcPod.Name(), dstPod.Name())
+
+	for {
+		// Don't retry lookups more often than once per second.
+		r := time.After(time.Second)
+
+		// we don't care about the actual response, we just want to check that DNS operativity.
+		target := "localhost"
```

Comment:
nit: Maybe add a comment that `localhost` is required here for the `local` plugin, since it might not be obvious to a casual reader